libethash-cuda/ethash_cuda_miner.cpp

```diff
@@ -303,7 +303,12 @@ void ethash_cuda_miner::search(uint8_t const* header, uint64_t target, search_ho
 		}
 		run_ethash_search(s_gridSize, s_blockSize, m_sharedBytes, stream, buffer, m_current_nonce, m_parallelHash);
 		if (m_current_index >= s_numStreams)
-			exit = (found_count && hook.found(nonces, found_count)) || hook.searched(batch_size);
+		{
```

Comment:
Excellent. searched was not called if solutions found.

Reply:
Yes, I noticed that during rebase. It looks I'm not the best reviewer :)

Reply:
Bingo. That would explain the slight hash rate loss with my previous changes. I had changed the bitwise or in the expression to a logical or, since both functions return bool. The bitwise or prevented the compiler from bypassing the searched function call if a valid solution was found. The searched method is where we accumulate hash count.

Great! These unexplained phenomenon really bug me.